installer.sh

```diff
@@ -59,33 +68,79 @@ confirm_force_install () {
-    PYTHON_PIPS="m2crypto"
-    BUILD_TOOLS="build-essential libtool automake pkg-config m4 libgcrypt20-dev uthash-dev autoconf autoconf-archive libcurl4-gnutls-dev gnulib doxygen libdbus-1-dev"
-    $PACKAGE_MGR update
-else
-   echo "No recognized package manager found on this system!" 1>&2
-   exit 1
+# Determine distibution (using systemd standard `os-release`):
+if [ -f /etc/os-release ]; then
+        . /etc/os-release
+    else
+        echo "Not able to determine your OS or Distribution"
+        exit 1
 fi
 
+case "$ID" in
+    debian | ubuntu)
+        echo "${ID} selected."
+        PACKAGE_MGR=$(command -v apt-get)
+        PYTHON_PREIN="git patch"
+        PYTHON_DEPS="python3 python3-pip python3-dev python3-setuptools python3-zmq python3-tornado python3-cryptography python3-simplejson python3-requests gcc g++ libssl-dev swig python3-yaml wget"
+        PYTHON_PIPS="m2crypto"
+        BUILD_TOOLS="build-essential libtool automake pkg-config m4 libgcrypt20-dev uthash-dev autoconf autoconf-archive libcurl4-gnutls-dev gnulib doxygen libdbus-1-dev"
+        NEED_BUILD_TOOLS=1
+        $PACKAGE_MGR update
+    ;;
+
+    rhel | centos)
+        case "${VERSION_ID}" in 
+            7)
+                echo "${ID} ${VERSION_ID} selected."
+                PACKAGE_MGR=$(command -v yum)
+                NEED_EPEL=1
+                PYTHON_PREIN="python36 python36-devel python36-setuptools python36-pip git wget patch openssl"
+                PYTHON_DEPS="gcc gcc-c++ openssl-devel swig python36-PyYAML python36-tornado python36-simplejson python36-cryptography python36-requests python36-zmq yaml-cpp-devel"
+                PYTHON_PIPS="m2crypto"
+                BUILD_TOOLS="openssl-devel file libtool make automake m4 libgcrypt-devel autoconf autoconf-archive libcurl-devel libstdc++-devel uriparser-devel dbus-devel gnulib-devel doxygen"
+                NEED_BUILD_TOOLS=1
+                CENTOS7_TSS_FLAGS="--enable-esapi=no --disable-doxygen-doc"
+            ;;
+            8)
+                echo "${ID} ${VERSION_ID} selected."
+                PACKAGE_MGR=$(command -v dnf)
+                NEED_EPEL=1
+                PYTHON_PREIN="python3 python3-devel python3-setuptools python3-pip"
+                PYTHON_DEPS="gcc gcc-c++ openssl-devel python3-yaml python3-requests swig python3-cryptography wget git"
+                PYTHON_PIPS="tornado==5.0.2 pyzmq m2crypto simplejson"
+                BUILD_TOOLS="git wget patch libyaml openssl-devel libtool make automake m4 libgcrypt-devel autoconf libcurl-devel libstdc++-devel dbus-devel"
+                #TPM2_TOOLS_PKGS="tpm2-tss tpm2-tools tpm2-abrmd" TODO: still on 3.1.1 tpm2_tools 
+                NEED_BUILD_TOOLS=1
+                NEED_PYTHON_DIR=1
+                POWERTOOLS="--enablerepo=PowerTools install autoconf-archive"
```

Comment:
_PowerTools_ might be CentOS-only.  I can't tell how to install the RHEL _CodeReady Builder_ repo, though it sounds like it is different.

Reply:
ok, i don't have an rhel 8 test system.  please update if you figure out how to get it on there.  I did smoke test this in centos 8 and it works.